spark3-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ExtendedBatchScanExec.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.v2
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.AttributeReference
+import org.apache.spark.sql.catalyst.plans.QueryPlan
+import org.apache.spark.sql.connector.read.{InputPartition, PartitionReaderFactory, Scan}
+
+// The only reason we need this class and cannot reuse BatchScanExec is because
+// BatchScanExec caches input partitions and we cannot apply file filtering before execution
+// Spark calls supportsColumnar during physical planning which, in turn, triggers split planning
+// We must ensure the result is not cached so that we can push down file filters later
+// The only difference compared to BatchScanExec is that we are using def instead of lazy val for splits
+case class ExtendedBatchScanExec(
```

Comment:
Maybe NonCachingBatchScanExec?

Reply:
I tend to name things with `Extended` prefix where we don't plan to have separate classes in the future.

Reply:
Ideally, we should not have this node at all.

Reply:
Should we try to get this fixed in Spark 3.1?

Reply:
I'd love too but we will have to justify it quite a bit. Do you think it is worth trying?